src/shared/components/selected-species/SelectedSpecies.tsx

```diff
@@ -34,7 +34,7 @@ export type Props = {
 
 const SelectedSpecies = (props: Props) => {
   const onClick = () => {
-    props.onClick ?? props.species;
+    props.onClick ? props.onClick(props.species) : null;
```

Comment:
I think what you are looking for is just `props.onClick?.(props.species)` instead of this whole line.